openshift/dynamic/client.py

```diff
@@ -878,12 +887,16 @@ def __search(self, parts, resources):
             elif isinstance(resourcePart, dict):
                 return self.__search(parts[1:], resourcePart)
             else:
+                if isinstance(resourcePart, ResourceList):
```

Comment:
I'm not sure I understand what this if block is doing, why would you need to wrap this in a list value only when it's a `ResourceList`?

Reply:
That's because you add `ResourceList` to resources as non-list on line 645:

```python
groups[DISCOVERY_PREFIX] = {'': {
            'v1': ResourceGroup(True, resources = {"List": ResourceList(self.client)})
        }}
```

Maybe it would be more reasonable to remove this condition, and change the 645 line to

```python
groups[DISCOVERY_PREFIX] = {'': {
            'v1': ResourceGroup(True, resources = {"List": [ResourceList(self.client)]})
        }}
```

Reply:
What do you think?

Reply:
Ah I see, that makes sense. I think it would be better to change 645 and have a consistent data structure

Reply:
Done.